Update parent pom version

vinayvishal opened the pull request:
Updated required dependency versions.
Added nexus-staging plugin.
Updated parent pom version to 1.0.5

Signed-off-by: Vinay Vishal <vinay.vishal@oracle.com>

bshannon commented on pom.xml:
```diff
@@ -104,6 +104,15 @@
                         </exclude>
                     </configuration>
                 </plugin> 
+                <plugin>
```
I don't think this change is needed when using the 1.0.5 parent pom.

romain-grecourt replied:
correct.

vinayvishal replied:
Done.

romain-grecourt commented on pom.xml:
```diff
@@ -19,6 +19,12 @@
+    <parent>
+        <groupId>org.eclipse.ee4j</groupId>
+        <artifactId>project</artifactId>
+        <version>1.0.5</version>
```
Use <relativePath/> to avoid the warning about the parent pom not available at ../pom.xml

vinayvishal replied:
Done.

Commit:
Update parent pom version

Updated required dependency versions.
Updated parent pom version to 1.0.5
Added parent pom to root pom

Signed-off-by: Vinay Vishal <vinay.vishal@oracle.com>
